functions/billing/index.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright 2018, Google LLC.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START functions_billing_limit]
+// [START functions_billing_stop]
+const {google} = require('googleapis');
+const {auth} = require('google-auth-library');
+
+const PROJECT_ID = process.env.GCP_PROJECT;
+const PROJECT_NAME = `projects/${PROJECT_ID}`;
+// [END functions_billing_stop]
+// [END functions_billing_limit]
+
+// [START functions_billing_slack]
+const slack = require('slack');
+
+const BOT_ACCESS_TOKEN = 'xxxx-111111111111-abcdefghidklmnopq';
+const CHANNEL = 'general';
+
+exports.notifySlack = async (data, context) => {
+  const pubsubMessage = data;
+  const budgetNotificationText = JSON.stringify(pubsubMessage.attributes) +
+    ', ' +
+    Buffer.from(pubsubMessage.data, 'base64').toString();
+
+  const res = await slack.chat.postMessage({
+    token: BOT_ACCESS_TOKEN,
+    channel: CHANNEL,
+    text: budgetNotificationText
+  });
+  console.log(res);
+};
+// [END functions_billing_slack]
+
+// [START functions_billing_stop]
+const billing = google.cloudbilling('v1').projects;
+
+exports.stopBilling = async (data, context) => {
+  const pubsubData = JSON.parse(Buffer.from(data.data, 'base64').toString());
+  if (pubsubData.costAmount <= pubsubData.budgetAmount) {
```

Comment:
Do we need to assure that costAmount is a number and not a string? Want to avoid cases like `'100' < '2' = true`

Reply:
FWIW this is how the code [is currently structured](https://cloud.google.com/billing/docs/how-to/notify#write_a_cloud_function_1).